nncf/quantization/algorithms/accuracy_control/algorithm.py

```diff
@@ -0,0 +1,342 @@
+                all_reverted_ops.difference_update(current_group.operations)
+                if exclude_bad_nodes:
+                    excluded_nodes.extend(current_group.quantizers)
+                    nncf_logger.debug('Quantizers were added to the excluded list: '
+                                      f'{_create_message(current_group.quantizers)}')
+                is_step_back = True
+
+            previous_accuracy_drop = current_accuracy_drop
+
+            nncf_logger.info('Re-calculating ranking scores for remaining groups')
+            ranker.rank_groups_of_quantizers(ranked_groups, initial_model, current_model, quantized_model_graph)
+
+        # Show results that were achieved.
+        if removed_all or not reached_required_drop:
+            nncf_logger.info('The algorithm could not achieve the required accuracy drop.', force=True)
+
+        if iteration + 1 >= self.max_num_iterations:
+            nncf_logger.info('Maximum number of iteration was reached.')
+
+        if not removed_all:
+            nncf_logger.debug(f'Quantizers that were removed:\n{_create_message(all_removed_nodes)}')
+            nncf_logger.info(f'{len(all_reverted_ops)} out of {num_of_quantized_ops} '
+                             f'were reverted back to the {precision_change_to} precision:'
+                             f'\n{_create_message(all_reverted_ops)}')
+
+        return current_model
+
+    # TODO(andrey-churkin): Should be removed when native implementation will become the main one.
+    @staticmethod
+    def _match_const_nodes_names(initial_model_graph: NNCFGraph,
+                                 quantized_model_graph: NNCFGraph,
+                                 const_metatypes: List[OperatorMetatype]) -> None:
+        initial_graph_const_nodes = initial_model_graph.get_nodes_by_metatypes(const_metatypes)
+        quantized_graph_const_nodes = quantized_model_graph.get_nodes_by_metatypes(const_metatypes)
+        for initial_graph_const_node in initial_graph_const_nodes:
+            num_matches = 0
+            for quantized_graph_const_node in quantized_graph_const_nodes:
+                if quantized_graph_const_node.node_name.startswith(initial_graph_const_node.node_name):
+                    assert quantized_graph_const_node.node_type == initial_graph_const_node.node_type
+                    quantized_graph_const_node.data[NNCFGraph.NODE_NAME_ATTR] = initial_graph_const_node.node_name
+                    num_matches += 1
+            assert num_matches == 1
+
+    @staticmethod
+    def _collect_original_biases_and_weights(initial_model_graph: NNCFGraph,
+                                             quantized_model_graph: NNCFGraph,
+                                             initial_model,
+                                             algo_backend: AccuracyControlAlgoBackend) -> None:
+        nodes_with_bias = (node for node in quantized_model_graph.get_all_nodes() \
```

Comment:
> Why get_all_nodes instead of get_nodes_by_metatypes? Wouldn't it save computational time for huge networks? The same question for the weights below.

The time complexity for both methods is the same.

> Also, can we fill the weights and biases in the one pass through the nodes list?

Done